crates/oxc_semantic/src/module_record/builder.rs

```diff
@@ -241,17 +241,27 @@ impl ModuleRecordBuilder {
         self.add_default_export(exported_name.span());
 
         let id = match &decl.declaration {
-            ExportDefaultDeclarationKind::Expression(_) => None,
-            ExportDefaultDeclarationKind::FunctionDeclaration(func) => func.id.as_ref(),
-            ExportDefaultDeclarationKind::ClassDeclaration(class) => class.id.as_ref(),
+            ExportDefaultDeclarationKind::Expression(expr) => {
+                if let Expression::Identifier(ident) = &expr {
+                    Some((ident.name.clone(), ident.span))
+                } else {
+                    None
+                }
+            }
```

Comment:
To solve this.
```ts
const a = 1;
export default a;
```

Reply:
We need to take a look at the spec on this one. The algorithms of this module strictly follow the spec.

Reply:
Ok, I will take a look

Reply:
What is the purpose of this change?
It should be `default` exported instead of `a`.

Reply:
> What is the purpose of this change? It should be `default` exported instead of `a`.

We need to add `SymbolFlags::Export` to all exports if the export is `Identifier` or `IdentifierReference `. If we want to solve this problem via `ModuleRecord` we must get all ExportEntries, But if the export entry is default we cannot get the `a` symbol currently, Because `ExportLocalName::Default` only have a span.

Reply:
I will revert these changes. I found the best way. I can get the symbol id via `symbols.get_symbol_id_from_span`

Reply:
The three exports below differ in their behavior: only 'a' and 'b' are live bindings exports, whereas 'c' is a value export.

```javascript
export function a() {}
export class b {}
export default c;
```

So, it is useful to track references to both `a` and `b`.
However, I don't see any *value* to track `c`

Reply:
I've already mentioned why we need to add SymbolFlags::Export here. https://github.com/oxc-project/oxc/pull/2508#issuecomment-1963253987